file/file_client.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2019 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package file
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/golang/protobuf/proto"
+	monitoring "google.golang.org/genproto/googleapis/monitoring/v3"
+)
+
+// FileClient allows writing to a file gRPC endpoint. The
```

Comment:
It's no clear what "the implementation may ..." means. I think you're documenting how this is used, which is less useful than documenting the requirements. Are you saying this needs to be the thread-safe?

Reply:
Sorry the comment here was not right at all. The newly created struct is under https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/124/files#diff-b8ca93bc37a9129de4e5193030dc2e3cR30 . 

The requirement for the corresponding factory is at : 
https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/124/files#diff-b1fc947b52e454cdf0a8ef16813d0802R612 

The topic about thread-safe is documented in Factory's New(): https://github.com/Stackdriver/stackdriver-prometheus-sidecar/pull/124/files#diff-b1fc947b52e454cdf0a8ef16813d0802R627